src/odemis/gui/model/tab_gui_data.py

```diff
@@ -476,7 +481,80 @@ def __init__(self, main):
+
+
+class CryoTdctCorrelationGUIData(CryoGUIData):
+    """ Represent an interface used to correlate using 3DCT.
+
+    Used for METEOR systems.
+
+    """
+
+    def __init__(self, main):
+        super().__init__(main)
+
+        # Current tool selected (from the toolbar)
+        tools = {TOOL_NONE, TOOL_RULER, TOOL_FIDUCIAL, TOOL_REGION_OF_INTEREST, TOOL_SURFACE_FIDUCIAL}
+        # Update the tool selection with the new tool list
+        self.tool.choices = tools
+
+        # the streams to correlate among all streams in .streams
+        self.selected_stream = model.VigilantAttribute(None)
+        # point on FIB stream to locate sample surface
+        self.fib_surface_point = model.VigilantAttribute(None)
+        # Output of 3DCT
+        self.projected_points: List[Target] = []
+
+        # for export tool
+        self.acq_fileinfo = VigilantAttribute(None)  # a FileInfo
+
+    def add_new_target(self, x, y, type, z=None):
+        """Targets added when tools in toolbox bar are toggled or when keyboard shortcuts are used.
+        :param x: (float) x position of the target
+        :param y: (float) y position of the target
+        :param type: (str) type of the given target like Fiducial, PointOfInterest or SurfaceFiducial
+        :param z: (float) optional z position for the given target. For target in FM, z is compulsory and for the target
+         in FIB, z is None.
+        :return: (Target or error) Target if parameters are valid otherwise logs error.
+        """
+
+        fm_focus_position = self.main.focus.position.value['z']
+        existing_names = [str(f.name.value) for f in self.main.targets.value]
+        pattern = r"-(\d+)"
+
+        if self.views.value[0] == self.focussedView.value: # FM view
+            if type == FIDUCIAL:
+                t_name = make_unique_name("FM-1", existing_names)
+                index = int(re.search(pattern, t_name).group(1))
+                target = Target(x, y, z=0, name=t_name, type=type,
+                                        index=index, fm_focus_position=fm_focus_position)
+
+            elif type == POI:
+                target = Target(x, y, z=0, name="POI-1", type=type,
+                                        index=1, fm_focus_position=fm_focus_position)
+
+        elif self.views.value[1] == self.focussedView.value: # FIB view
+            if type == SURFACE_FIDUCIAL:
+                target = Target(x, y, z=0, name="FIB_surface", type=type, index=1,
+                                fm_focus_position=fm_focus_position)
+                self.fib_surface_point.value = target
+                return target
+            t_name = make_unique_name("FIB-1", existing_names)
+            index = int(re.search(pattern, t_name).group(1))
+            target = Target(x, y, z=0, name=t_name, type=type, index=index,
+                            fm_focus_position=fm_focus_position)
+
+        else:
+            raise ValueError("Invalid selected view")
```

Comment:
Why raising an error in such case, instead of just logging + returning `None`? Most likely the caller is not at fault (directly).